pkg/util/deployment/deployment.go

```diff
@@ -367,6 +374,19 @@ func NewRSNewReplicas(deployment *extensions.Deployment, allRSs []*extensions.Re
 	}
 }
 
+// IsSaturated checks if the new replica set is saturated by comparing its size with its deployment size.
+func IsSaturated(deployment *extensions.Deployment, rs *extensions.ReplicaSet) bool {
+	if rs == nil {
+		return false
+	}
+	desiredString := rs.Annotations[DesiredReplicasAnnotation]
+	desired, err := strconv.Atoi(desiredString)
+	if err != nil {
+		return false
+	}
+	return rs.Spec.Replicas == deployment.Spec.Replicas && int32(desired) == deployment.Spec.Replicas
```

Comment:
what does it mean when int32(desired) == deployment.Spec.Replicas?  Does this mean that data is not consistent between what the deployment thinks the total replica count is and the replica set thinks it should be?  Is this caused by a failed write or a race condition between updating the Dep and RS?